### ISSUE TITLE
table schema constraints inefficient

### ISSUE THREAD

**Issue by micimize** [frictionlessdata/goodtables-py#302]
Every table schema based constraint calls `field.test_value(value, constraints=[constraint])`. This results in a lot of avoidable overhead. Instead, we could:
* add the constraint name as a field on `CastError`
* consolidate the constraints into a single call (maybe in `compile_checks`)
* add an adapter/handler per-constraint

**Comment by micimize** [comment]
I've [implemented the above](https://gist.github.com/micimize/a96dbebe5365cc158b5589e1f6f461a4) by extending `tableschema.Field` and accruing failed constraints. Type errors swallow other constraints.

In the context of the program I'm profiling, the improvement was significant. A real PR will still need to validate results consistency and maybe handle the consolidation internally
```
constraints_checks:11:_check_constraint
 total time: 35.54%
  self time: (5.52%)
total calls: 60048×

vs

schema_constraints_checks:29:schema_constraints_checks
 total time: 13.60%
  self time: (1.61%)
total calls: 10008×
```

**Comment by roll** [comment]
@micimize 
I agree that's a problem

**Comment by roll** [comment]
MERGED into #341
